app/src/main/java/com/example/cocktailmaster/ui/home/HomeFragment.kt

```diff
@@ -1,28 +1,45 @@
 package com.example.cocktailmaster.ui.home
 
 import android.view.View
+import androidx.appcompat.widget.SearchView
+import com.example.cocktailmaster.R
 import com.example.cocktailmaster.base.BaseFragment
 import com.example.cocktailmaster.data.model.Drink
+import com.example.cocktailmaster.data.repository.DrinksRemoteRepository
+import com.example.cocktailmaster.data.source.remote.SearchDrinkDataSource
 import com.example.cocktailmaster.databinding.FragmentHomeBinding
 import com.example.cocktailmaster.ui.RepositoryUtils
 import com.example.cocktailmaster.ui.home.adapter.AlphabetAdapter
+import com.example.cocktailmaster.ui.home.adapter.DrinksHintAdapter
+import com.example.cocktailmaster.ui.home.adapter.RandomDrinksAdapter
 
 class HomeFragment : BaseFragment<FragmentHomeBinding>(FragmentHomeBinding::inflate),
-    HomeContract.View {
+    HomeContract.View,
+    View.OnClickListener,
+    SearchView.OnCloseListener,
+    SearchView.OnQueryTextListener {
 
     private var presenter: HomePresenter? = null
     private val drinkAdapter = RandomDrinksAdapter()
     private val alphabetAdapter = AlphabetAdapter()
+    private val drinksHintAdapter = DrinksHintAdapter()
     private val drinks = mutableListOf<Drink>()
 
     override fun initViews() {
+        listOf(binding.imageSearch).forEach {
+            it.setOnClickListener(this)
+        }
         binding.recyclerDrinks.adapter = drinkAdapter
         binding.includeAlphabets.recyclerAlphabets.adapter = alphabetAdapter
+        binding.recyclerDrinksHint.adapter = drinksHintAdapter
     }
 
     override fun initData() {
-        val randomDrinksRepo = RepositoryUtils.getRandomDrinksRepo()
-        presenter = HomePresenter(this, randomDrinksRepo)
+        binding.searchDrinks.also {
```

Comment:
sao k dùng apply, run hay j đó mà lại dùng also để r lại thêm cái it kia nhỉ

Reply:
e thấy dùng run với apply thì this bên trong nó bị sai context
![image](https://user-images.githubusercontent.com/53604984/119472301-541d6280-bd74-11eb-92ea-ea2fb10ce5a6.png)

Reply:
this@Accitviy hay là this@Fragment ......, chứ kia n đang hiểu là searchview